projects/packages/my-jetpack/_inc/components/golden-token/index.jsx

```diff
@@ -0,0 +1,167 @@
+import { Modal } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React, { useCallback, useRef, useState } from 'react';
+import { JetpackBackupLogo, JetpackLogo, JetpackScanLogo } from './logos';
+import styles from './styles.module.scss';
+
+/**
+ * Component that displays a golden token experience.
+ *
+ * @param {object} props - Component props.
+ * @param {Function} props.redeemClick - Callback function to handle redeem click.
+ * @param {object} props.userConnectionData - Connected user data.
+ * @param {Function} props.onClickGoBack - Callback function to handle go back click.
+ * @param {Array} props.purchases - Array of Object of purchases.
+ * @param {boolean} props.fetchingPurchases - Are purchases being fetched.
+ * @returns {React.Component} - GoldenToken component.
+ */
+function GoldenTokenModal( {
+	redeemClick,
+	userConnectionData,
+	onClickGoBack,
+	fetchingPurchases,
+	purchases,
+} ) {
+	const [ hasAnimated, setIsAnimating ] = useState( false );
+	const videoRef = useRef( null );
+
+	// Any purchase with the partner_slug of 'goldenticket' is considered a golden token.
+	const goldenToken = purchases.filter( golden => golden.partner_slug === 'goldenticket' );
+	const hasGoldenToken = goldenToken.length > 0;
+
+	const redeemClickHandler = useCallback(
+		e => {
+			redeemClick?.( e );
+			setIsAnimating( true );
+			videoRef.current.play();
+		},
+		[ videoRef, redeemClick ]
+	);
+
+	const maybeReanimate = useCallback( () => {
+		hasAnimated && videoRef.current.play();
+	}, [ hasAnimated ] );
+
+	const modalClassName = classNames( styles.modal, {
+		[ styles.animating ]: hasAnimated,
+	} );
+
+	// They might not have a display name set in wpcom, so fall back to wpcom login or local username.
+	const wpcomUserName =
+		userConnectionData?.currentUser?.wpcomUser?.display_name ||
+		userConnectionData?.currentUser?.wpcomUser?.login ||
+		userConnectionData?.currentUser?.username;
+
+	return (
+		<Modal className={ modalClassName } onRequestClose={ onClickGoBack } isDismissible={ false }>
+			{ fetchingPurchases && <>{ __( 'Checking gold status…', 'jetpack-my-jetpack' ) }</> }
+			{ ! fetchingPurchases && hasGoldenToken && (
+				<Container horizontalSpacing={ 3 } horizontalGap={ 3 }>
+					<Col>
+						<JetpackLogo className={ styles[ 'jetpack-logo' ] } />
+						<div
+							className={ styles[ 'video-wrap' ] }
+							onClick={ maybeReanimate }
+							role="presentation"
+						>
+							{ /* eslint-disable-next-line jsx-a11y/media-has-caption */ }
+							<video
+								ref={ videoRef }
+								src="https://videos.files.wordpress.com/oSlNIBQO/jetpack-golden-token.mp4"
+							/>
+						</div>
+
+						<div className={ styles[ 'content-wrap' ] }>
+							<div className={ styles[ 'content-wrap-text' ] }>
+								<p className={ styles[ 'hi-user' ] }>Hey, { wpcomUserName }</p>
```

Comment:
Should we make this translatable?

```suggestion
								<p className={ styles[ 'hi-user' ] }>
									{ sprintf(
										/* Translators: %s is the user's display name. */
										__( 'Hey, %s', 'jetpack-my-jetpack' ),
										wpcomUserName
									) }
								</p>
```